magicctapipe/scripts/mars/tests/compare_hillas_stereo_params.py

```diff
@@ -0,0 +1,291 @@
+
+    # LON_ORM = u.Quantity(-17.89064, u.deg)
+    # LAT_ORM = u.Quantity(28.76177, u.deg)
+    # HEIGHT_ORM = u.Quantity(2199.835, u.m)
+
+    # location = EarthLocation.from_geodetic(lon=LON_ORM, lat=LAT_ORM, height=HEIGHT_ORM)
+    # times1 = [str(float(sec))[:-2] for sec in df_mcp['time_sec']]
+    # times2 = [str(float(nano))[:-2] for nano in df_mcp['time_nanosec']]
+    # times = [f"{sec}.{nanosec}" for sec, nanosec in zip(times1, times2)]
+    #
+    # event_times = Time(times, format="unix", scale="utc")
+    # source_coords = SkyCoord(ra=83.6333 * u.degree, dec=22.0133 * u.degree, frame='icrs')
+    # la_palma = AltAz(location=location, obstime=event_times)
+    # source_coords_altaz = source_coords.transform_to(la_palma)
+    #
+    # theta = angular_separation(
+    # lon1=u.Quantity(df_mcp['az'].to_numpy(), u.deg),
+    # lat1=u.Quantity(df_mcp['alt'].to_numpy(), u.deg),
+    # lon2=source_coords_altaz.az,
+    # lat2=source_coords_altaz.alt,
+    # )
+    #
+    # df_mcp['theta2'] = theta.to(u.deg).value ** 2
+    # df_mcp['theta'] = theta.to(u.deg).value
+
+    # apply cuts
+    # 50<Size<50000, leakage<0.15, impact<120, theta<0.14
+    # df_mcp_cut = df_mcp.loc[(df_mcp["intensity"] < 50000) & (df_mcp["intensity"] > 50) & (df_mcp["intensity_width_1"] < 0.15) & (df_mcp['theta'] < 0.14) & (df_mcp["impact"] < 120)]
+    df_mcp_cut = df_mcp.loc[
+        (df_mcp["intensity"] < 50000)
+        & (df_mcp["intensity"] > 50)
+        & (df_mcp["intensity_width_1"] < 0.15)
+    ]
+
+    # filter for M1 or M2
+    df_mcp_m1 = df_mcp_cut[df_mcp_cut["tel_id"] == 2]
+    df_mcp_m2 = df_mcp_cut[df_mcp_cut["tel_id"] == 3]
+
+    print(f"Number of M1 events after cut {df_mcp_m1.shape[0]}")
+    print(f"Number of M2 events after cut {df_mcp_m2.shape[0]}")
+
+    # ---------------
+    # read MARS data
+    # ---------------
+    # contains values for both M1 and M2
+    cut_params = [
+        "size_M1",
+        "size_M2",
+        "leakage_M1",
+        "leakage_M2",
+        "impact_M1",
+        "impact_M2",
+    ]
+    df_mars = pd.DataFrame()
+    with uproot.open(config["MARS-input"]["MARS-path"]) as file_mars:
+        for cut_par in cut_params:
+            mars_name = mars_params[cut_par]
+            df_mars["event_id"] = file_mars["Events"][
+                "MRawEvtHeader_1.fStereoEvtNumber"
+            ].array(library="np")
+            df_mars["mars_" + cut_par] = file_mars["Events"][mars_name].array(
+                library="np"
+            )
+        for par in config["Params"]:
+            mars_name = mars_params[par]
+            df_mars["mars_" + par] = file_mars["Events"][mars_name].array(library="np")
+
+    # apply cuts
+    # 50<Size<50000, leakage<0.15, impact<120
+    df_mars_cut = df_mars.loc[
+        (df_mars["mars_size_M1"] < 50000)
+        & (df_mars["mars_size_M2"] < 50000)
+        & (df_mars["mars_size_M1"] > 50)
+        & (df_mars["mars_size_M2"] > 50)
+        & (df_mars["mars_leakage_M1"] < 0.15)
```

Comment:
leakage cuts are applied in MARS in the training, but the events are not removed later on in the  analysis (nevertheless often will have a high hadronness and thus would not survive the cuts)